windows/test-app.ts

```diff
@@ -0,0 +1,300 @@
+import * as path from 'path';
+import * as fs from 'fs';
+
+const windowsDir = 'windows';
+const reactNativeModulePath = findClosestPathTo(path.join('node_modules', 'react-native-windows'));
+const testAppNodeModulePath = findClosestPathTo(path.join('node_modules', 'react-native-test-app'));
+const srcRootPath = path.join(testAppNodeModulePath, windowsDir);
+const destPath = path.resolve('');
+
+
+function findClosestPathTo(fileOrDirName: string): string {
+    var basePath = path.resolve('');
+    var rootDirectory = basePath.split(path.sep)[0] + path.sep;
+
+    while (basePath !== rootDirectory) {
+        const candidatePath = path.join(basePath, fileOrDirName)
+        if (fs.existsSync(candidatePath)) {
+            return path.relative('', candidatePath);
+        }
+        //Get parent folder
+        basePath = path.dirname(basePath)
+    }
+
+    return null
+}
+
+/**
+ * Text to replace, + config options
+ */
+export type Replacements = {
+    regExpPatternsToRemove?: RegExp[];
+    [key: string]: any;
+};
+
+function walk(current: string): string[] {
+    if (!fs.lstatSync(current).isDirectory()) {
+        return [current];
+    }
+
+    const files = fs
+        .readdirSync(current)
+        .map(child => walk(path.join(current, child)));
+    const result: string[] = [];
+    return result.concat.apply([current], files);
+}
+
+/**
+ * Get a source file and replace parts of its contents.
+ * @param srcPath Path to the source file.
+ * @param replacements e.g. {'TextToBeReplaced': 'Replacement'}
+ * @return The contents of the file with the replacements applied.
+ */
+export function resolveContents(
+    srcPath: string,
+    replacements: Replacements,
+): string {
+    let content = fs.readFileSync(srcPath, 'utf8');
+
+    Object.keys(replacements).forEach(regex => {
+        content = content.replace(new RegExp(regex, 'g'), replacements[regex]);
+    });
+
+    return content;
```

Comment:
This can be simplified further by using `reduce`:
```suggestion
    const content = fs.readFileSync(srcPath, 'utf8');
    return Object.keys(replacements).reduce((regex) => {
        return content.replace(new RegExp(regex, 'g'), replacements[regex]);
    }, content);
```